AOT compilation with allowJs: true fails in 5.0.0-rc.2

<!--
PLEASE HELP US PROCESS GITHUB ISSUES FASTER BY PROVIDING THE FOLLOWING INFORMATION.

ISSUES MISSING IMPORTANT INFORMATION MAY BE CLOSED WITHOUT INVESTIGATION.
-->

## I'm submitting a...
<!-- Check one of the following options with "x" -->
<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  <!-- Please search GitHub for a similar issue or PR before submitting -->
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
<!-- Describe how the issue manifests. -->
AOT compilation fails with `"allowJs": true`.


## Expected behavior
<!-- Describe what the desired behavior would be. -->
AOT compilation works with `"allowJs": true`.


## Minimal reproduction of the problem with instructions
```
git clone https://github.com/filipesilva/ngc-allowjs
cd ngc-allowjs
npm i
npm run ngc # for ngc failure
npm run ng -- build --aot # for angular cli failure
# remove "allowJs": false from src/tsconfig.app.json to make these builds work again
```

## What is the motivation / use case for changing the behavior?
<!-- Describe the motivation or the concrete use case. -->


## Environment

<pre><code>
Angular version: 1.5.0-rc.2
<!-- Check whether this is still an issue in the most recent Angular version -->

Browser:
- [x] Chrome (desktop) version 61
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: XX  6.11.1
- Platform:  Windows 10

Others:
<!-- Anything else relevant?  Operating system version, IDE, package manager, HTTP server, ... -->
</code></pre>

Compilation fails in different ways with `ngc` or Angular CLI.

For `ngc`:
```
kamik@T460p MINGW64 /d/sandbox/ngc-allowjs
$ npm run ngc

> master-project@0.0.0 ngc D:\sandbox\ngc-allowjs
> ngc -p src/tsconfig.ngc.json

Error: TypeError: Cannot read property 'type' of undefined
    at Object.getEffectiveTypeAnnotationNode (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:9432:17)
    at assignContextualParameterTypes (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:37839:25)
    at checkFunctionExpressionOrObjectLiteralMethod (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:38136:29)
    at checkExpressionWorker (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:39148:28)
    at checkExpression (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:39077:42)
    at checkBinaryLikeExpression (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:38657:29)
    at checkBinaryExpression (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:38649:20)
    at checkExpressionWorker (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:39169:28)
    at checkExpression (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:39077:42)
    at checkBinaryLikeExpression (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:38657:29)

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! master-project@0.0.0 ngc: `ngc -p src/tsconfig.ngc.json`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the master-project@0.0.0 ngc script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\kamik\AppData\Roaming\npm-cache\_logs\2017-10-17T10_14_11_075Z-debug.log
```

For Angular CLI:
```
kamik@T460p MINGW64 /d/sandbox/ngc-allowjs
$ npm run ng -- build --aot

> master-project@0.0.0 ng D:\sandbox\ngc-allowjs
> ng "build" "--aot"

Date: 2017-10-17T10:12:58.261Z
Hash: 80e92f8d9f5182151fab
Time: 5342ms
chunk {inline} inline.bundle.js, inline.bundle.js.map (inline) 5.83 kB [entry] [rendered]
chunk {main} main.bundle.js, main.bundle.js.map (main) 557 bytes [initial] [rendered]
chunk {polyfills} polyfills.bundle.js, polyfills.bundle.js.map (polyfills) 577 bytes {inline} [initial] [rendered]
chunk {styles} styles.bundle.js, styles.bundle.js.map (styles) 11.3 kB {inline} [initial] [rendered]

ERROR in ./src/main.ts
Module build failed: Error: TypeScript compilation failed.
    at plugin.done.then (D:\sandbox\ngc-allowjs\node_modules\@ngtools\webpack\src\loader.js:479:27)
    at process._tickCallback (internal/process/next_tick.js:109:7)
 @ multi ./src/main.ts
ERROR in ./src/polyfills.ts
Module build failed: Error: TypeScript compilation failed.
    at plugin.done.then (D:\sandbox\ngc-allowjs\node_modules\@ngtools\webpack\src\loader.js:479:27)
    at process._tickCallback (internal/process/next_tick.js:109:7)
 @ multi ./src/polyfills.ts
ERROR in error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/es6/reflect.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/es7/reflect.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_a-function.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_an-instance.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_an-object.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_array-from-iterable.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_array-includes.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_array-methods.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_array-species-constructor.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_array-species-create.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_bind.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_classof.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_cof.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_collection-strong.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_collection-weak.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_collection.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_core.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_ctx.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_defined.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_descriptors.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_dom-create.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_enum-bug-keys.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_export.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_fails.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_for-of.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_global.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_has.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_hide.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_html.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_ie8-dom-define.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_inherit-if-required.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_invoke.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_iobject.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_is-array-iter.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_is-array.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_is-object.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_iter-call.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_iter-create.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_iter-define.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_iter-detect.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_iter-step.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_iterators.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_library.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_meta.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_metadata.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_object-assign.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_object-create.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_object-dp.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_object-dps.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_object-gopd.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_object-gopn.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_object-gops.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_object-gpo.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_object-keys-internal.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_object-keys.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_object-pie.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_own-keys.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_property-desc.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_redefine-all.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_redefine.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_set-proto.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_set-species.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_set-to-string-tag.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_shared-key.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_shared.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_to-absolute-index.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_to-integer.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_to-iobject.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_to-length.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_to-object.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_to-primitive.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_uid.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_validate-collection.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/_wks.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/core.get-iterator-method.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es6.map.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es6.reflect.apply.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es6.reflect.construct.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es6.reflect.define-property.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es6.reflect.delete-property.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es6.reflect.enumerate.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es6.reflect.get-own-property-descriptor.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es6.reflect.get-prototype-of.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es6.reflect.get.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es6.reflect.has.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es6.reflect.is-extensible.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es6.reflect.own-keys.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es6.reflect.prevent-extensions.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es6.reflect.set-prototype-of.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es6.reflect.set.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es6.set.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es6.weak-map.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es7.reflect.define-metadata.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es7.reflect.delete-metadata.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es7.reflect.get-metadata-keys.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es7.reflect.get-metadata.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es7.reflect.get-own-metadata-keys.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es7.reflect.get-own-metadata.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es7.reflect.has-metadata.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es7.reflect.has-own-metadata.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/core-js/modules/es7.reflect.metadata.js' because it would overwrite input file.
error TS5055: Cannot write file 'D:/sandbox/ngc-allowjs/node_modules/zone.js/dist/zone.js' because it would overwrite input file.
Error: TypeError: Cannot read property 'type' of undefined
    at Object.getEffectiveTypeAnnotationNode (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:9432:17)
    at assignContextualParameterTypes (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:37839:25)
    at checkFunctionExpressionOrObjectLiteralMethod (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:38136:29)
    at checkExpressionWorker (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:39148:28)
    at checkExpression (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:39077:42)
    at checkBinaryLikeExpression (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:38657:29)
    at checkBinaryExpression (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:38649:20)
    at checkExpressionWorker (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:39169:28)
    at checkExpression (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:39077:42)
    at checkBinaryLikeExpression (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:38657:29)
    at checkBinaryExpression (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:38649:20)
    at checkExpressionWorker (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:39169:28)
    at checkExpression (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:39077:42)
    at checkExpressionStatement (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:41199:13)
    at checkSourceElement (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:43009:28)
    at Object.forEach (D:\sandbox\ngc-allowjs\node_modules\typescript\lib\typescript.js:1500:30)

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! master-project@0.0.0 ng: `ng "build" "--aot"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the master-project@0.0.0 ng script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\kamik\AppData\Roaming\npm-cache\_logs\2017-10-17T10_12_58_308Z-debug.log
```

Original issue in https://github.com/angular/angular-cli/issues/8061.

/cc @tbosch

@chuckjaz fix doesn't work for all cases. Still doesn't work if you have any javascript file in a project.

**Steps to reproduce:**
- generate new project using ng new
- just CREATE one js file in src/app
- run ng serve

Still getting error
`ERROR in error TS5055: Cannot write file 'MYDIR/test-app/src/app/test.js' because it would overwrite input file.
  Adding a tsconfig.json file will help organize projects that contain both TypeScript and JavaScript files. Learn more at https://aka.ms/tsconfig.`

So I am a bit confused now,  I am quite new to Angular, I thought the below issue was being dealt with? Also could someone explain if I was to use ng serve, that I could still work with external javascripst or will this not work due to this issue? 
I tried downgrading to angular 4 in the meantime, but I run in so many issues gyp-errors, ng path errors that I am starting to regret my decision to uninstall angular 5. 


ERROR in error TS5055: Cannot write file 'MYDIR/test-app/src/app/test.js' because it would overwrite input file. Adding a tsconfig.json file will help organize projects that contain both TypeScript and JavaScript files. Learn more at https://aka.ms/tsconfig.

I'm also getting this issue when running start after **ng eject**:
`ERROR in error TS5055: Cannot write file '<JS FILE PATH>' because it would overwrite input file.`

--- a/packages/compiler-cli/src/transformers/compiler_host.ts
+++ b/packages/compiler-cli/src/transformers/compiler_host.ts
@@ -130,7 +130,7 @@ export class TsCompilerAotCompilerTypeCheckHostAdapter implements ts.CompilerHos
                      moduleName, containingFile.replace(/\\/g, '/'), this.options, this,
                      this.moduleResolutionCache)
                    .resolvedModule;
-    if (rm && this.isSourceFile(rm.resolvedFileName)) {
+    if (rm && this.isSourceFile(rm.resolvedFileName) && DTS.test(rm.resolvedFileName)) {
       // Case: generateCodeForLibraries = true and moduleName is
       // a .d.ts file in a node_modules folder.
       // Need to set isExternalLibraryImport to false so that generated files for that file

--- a/packages/compiler-cli/test/ngc_spec.ts
+++ b/packages/compiler-cli/test/ngc_spec.ts
@@ -1647,6 +1647,38 @@ describe('ngc transformer command-line', () => {
       `);
          expect(main(['-p', path.join(basePath, 'src/tsconfig.json')])).toBe(0);
        });
+
+    it('should not type check a .js files from node_modules with allowJs', () => {
+      write('src/tsconfig.json', `{
+        "extends": "../tsconfig-base.json",
+        "compilerOptions": {
+          "noEmitOnError": true,
+          "allowJs": true,
+          "declaration": false
+        },
+        "files": ["test-module.ts"]
+      }`);
+      write('src/test-module.ts', `
+        import {Component, NgModule} from '@angular/core';
+        import 'my-library';
+
+        @Component({
+          template: 'hello'
+        })
+        export class HelloCmp {}
+
+        @NgModule({
+          declarations: [HelloCmp],
+        })
+        export class MyModule {}
+      `);
+      write('src/node_modules/t.txt', ``);
+      write('src/node_modules/my-library/index.js', `
+        export someVar = 1;
+        export someOtherVar = undefined + 1;
+      `);
+      expect(main(['-p', path.join(basePath, 'src/tsconfig.json')])).toBe(0);
+    });
   });
 
   describe('formatted messages', () => {

